qemu/tests/timerdevice_boot.py

```diff
@@ -42,6 +44,38 @@ def verify_guest_clock_source(session, expected):
             raise error.TestFail(
                 "Guest didn't use '%s' clocksource" % expected)
 
+    def verify_base_date_system_time(time_command, time_filter, time_format)
```

Comment:
You don't have to pass the variants, all of your functions is under same context.